### PR TITLE
some good utilties

### PR DESCRIPTION
**Pull request by zhenglaizhang**

### CODE REVIEW

File: util/src/main/scala/types/control/manage.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package net.zhenglai
+package types.control
+
+import scala.language.reflectiveCalls
+import scala.util.control.NonFatal
+
+object manage {
+  def apply[R <: { def close(): Unit }, B](resource: => R)(
```
Comment (coding2world) [inline util/src/main/scala/types/control/manage.scala:8]: 这个写法叫什么？

File: util/src/test/scala/types/controlTest.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    assert(lines.nonEmpty)
+  }
+
+  test("test continue") {
+    var count = 0
+    continue(count < 5) {
+      println(s"at $count")
+      count += 1
+    }
+    assert(count == 5)
```
Comment (coding2world) [inline util/src/test/scala/types/controlTest.scala:26]: 这里我理解是`body`代码块在每次执行的时候都改变了`cond`代码块引用的变量`count`。
就是`body`和`cond`使用`count`通信交流。

File: util/src/main/scala/types/control/manage.scala (hunk 1 above)
Comment (zhenglaizhang) [reply util/src/main/scala/types/control/manage.scala:8]: structural type.
It's used as some types doesn't share the same interface, but has similar methods structure. This mechanism uses reflection so pay attention to the performance overhead.

File: util/src/test/scala/types/controlTest.scala (hunk 2 above)
Comment (zhenglaizhang) [reply util/src/test/scala/types/controlTest.scala:26]: exactly

**Pull request merged**